packages/evm/src/evm.ts

```diff
@@ -617,8 +633,9 @@ export class EVM implements EVMInterface {
         // It is thus an unnecessary default item, which we have to save to dik
         // It does change the state root, but it only wastes storage.
         //await this._state.putContractCode(message.to, result.returnValue)
+        //const account = await this.eei.getAccount(message.to)
         const account = await this.eei.getAccount(message.to)
-        await this.eei.putAccount(message.to, account)
+        await this.eei.putAccount(message.to, account!)
```

Comment:
Ah ok, just re-read, this is exactly what is written in the comment (guess it's from you, right?), so that pre-Homestead there is explicitly an empty account created in this situation, have updated https://github.com/ethereumjs/ethereumjs-monorepo/pull/2634/commits/87e4541bb21cd5116fde0efe55b1b4b73d14965a.

Have directly pushed, interested in what this is doing with the tests.

Reply:
(unfortunately I didn't have the all-HFs label activated on that push. Now added so we'll see on push after that. Might eventually fix the remaining 6 blockchain tests? 🤔)